# Inicialización Grader


In [1]:
!wget --no-cache -O init.py -q https://raw.githubusercontent.com/rramosp/ai4eng.v1/main/content/init.py
import init; init.init(force_download=False); init.get_weblink()

replicating local resources


# Código

Código correspondiente a la carga, limpieza y preprocesado de los datos de entrenamiento del modelo.

## Archivos de datos

In [5]:
import os
os.environ['KAGGLE_CONFIG_DIR'] = '.'
!chmod 600 ./kaggle.json
!kaggle competitions download -c udea-ai-4-eng-20252-pruebas-saber-pro-colombia

  0% 0.00/29.9M [00:00<?, ?B/s]
100% 29.9M/29.9M [00:00<00:00, 1.40GB/s]


Se descomprime el archivo con los datos:

In [6]:
!unzip udea*.zip > /dev/null

Se hace un recuento preliminar del tamaño de los archivos:

In [8]:
!wc *.csv

   296787    296787   4716673 submission_example.csv
   296787   4565553  59185238 test.csv
   692501  10666231 143732437 train.csv
  1286075  15528571 207634348 total


## Librerías

In [10]:
import pandas as pd
import seaborn as sns
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

## Carga de datos

In [26]:
training_data = pd.read_csv("train.csv")
print ("Tamaño del DataFrame:", training_data.shape)

Tamaño del DataFrame:  (692500, 21)


Breve exploración de los datos y las columnas:

In [25]:
training_data.head(10)

ID  PERIODO_ACADEMICO                     E_PRGM_ACADEMICO  \
0  904256              20212                           ENFERMERIA   
1  645256              20212                              DERECHO   
2  308367              20203                MERCADEO Y PUBLICIDAD   
3  470353              20195           ADMINISTRACION DE EMPRESAS   
4  989032              20212                           PSICOLOGIA   
5  659872              20203                 MEDICINA VETERINARIA   
6   47159              20183                  INGENIERIA MECANICA   
7   11829              20183  ADMINISTRACIÓN EN SALUD OCUPACIONAL   
8  257869              20212                INGENIERIA INDUSTRIAL   
9  465511              20183           ADMINISTRACION DE EMPRESAS   

  E_PRGM_DEPARTAMENTO               E_VALORMATRICULAUNIVERSIDAD  \
0              BOGOTÁ  Entre 5.5 millones y menos de 7 millones   
1           ATLANTICO  Entre 2.5 millones y menos de 4 millones   
2              BOGOTÁ  Entre 2.5 millones y menos de 4 millones   
3           SANTANDER  Entre 4 millones y menos de 5.5 millones   
4           ANTIOQUIA  Entre 2.5 millones y menos de 4 millones   
5           ANTIOQUIA                         Más de 7 millones   
6               HUILA  Entre 2.5 millones y menos de 4 millones   
7              BOGOTÁ    Entre 1 millón y menos de 2.5 millones   
8           ATLANTICO  Entre 5.5 millones y menos de 7 millones   
9           ANTIOQUIA  Entre 2.5 millones y menos de 4 millones   

  E_HORASSEMANATRABAJA F_ESTRATOVIVIENDA F_TIENEINTERNET  \
0    Menos de 10 horas         Estrato 3              Si   
1                    0         Estrato 3              No   
2      Más de 30 horas         Estrato 3              Si   
3                    0         Estrato 4              Si   
4  Entre 21 y 30 horas         Estrato 3              Si   
5    Menos de 10 horas         Estrato 5              Si   
6  Entre 21 y 30 horas         Estrato 2              Si   
7  Entre 11 y 20 horas         Estrato 2              Si   
8    Menos de 10 horas         Estrato 1              Si   
9      Más de 30 horas         Estrato 5              Si   

                     F_EDUCACIONPADRE F_TIENELAVADORA  ... E_PRIVADO_LIBERTAD  \
0    Técnica o tecnológica incompleta              Si  ...                  N   
1      Técnica o tecnológica completa              Si  ...                  N   
2  Secundaria (Bachillerato) completa              Si  ...                  N   
3                             No sabe              Si  ...                  N   
4                   Primaria completa              Si  ...                  N   
5      Educación profesional completa              Si  ...                  N   
6    Educación profesional incompleta              Si  ...                  N   
7                 Primaria incompleta              Si  ...                  N   
8  Secundaria (Bachillerato) completa              Si  ...                  N   
9                           Postgrado              Si  ...                  N   

  E_PAGOMATRICULAPROPIO F_TIENECOMPUTADOR F_TIENEINTERNET.1  \
0                    No                Si                Si   
1                    No                Si                No   
2                    No                No                Si   
3                    No                Si                Si   
4                    No                Si                Si   
5                    No                Si                Si   
6                    Si                Si                Si   
7                    Si                No                Si   
8                    Si                Si                Si   
9                    Si                Si                Si   

                       F_EDUCACIONMADRE RENDIMIENTO_GLOBAL INDICADOR_1  \
0                             Postgrado         medio-alto       0.322   
1      Técnica o tecnológica incompleta               bajo       0.311   
2    Secundaria (Bachillerato) completa 

In [23]:
training_data.columns

Index(['ID', 'PERIODO_ACADEMICO', 'E_PRGM_ACADEMICO', 'E_PRGM_DEPARTAMENTO',
       'E_VALORMATRICULAUNIVERSIDAD', 'E_HORASSEMANATRABAJA',
       'F_ESTRATOVIVIENDA', 'F_TIENEINTERNET', 'F_EDUCACIONPADRE',
       'F_TIENELAVADORA', 'F_TIENEAUTOMOVIL', 'E_PRIVADO_LIBERTAD',
       'E_PAGOMATRICULAPROPIO', 'F_TIENECOMPUTADOR', 'F_TIENEINTERNET.1',
       'F_EDUCACIONMADRE', 'RENDIMIENTO_GLOBAL', 'INDICADOR_1', 'INDICADOR_2',
       'INDICADOR_3', 'INDICADOR_4'],
      dtype='object')

Tipos de datos de cada columna:

In [24]:
for c in training_data.columns:
    print ("%20s"%c, training_data[c].dtype)

                  ID int64
   PERIODO_ACADEMICO int64
    E_PRGM_ACADEMICO object
 E_PRGM_DEPARTAMENTO object
E_VALORMATRICULAUNIVERSIDAD object
E_HORASSEMANATRABAJA object
   F_ESTRATOVIVIENDA object
     F_TIENEINTERNET object
    F_EDUCACIONPADRE object
     F_TIENELAVADORA object
    F_TIENEAUTOMOVIL object
  E_PRIVADO_LIBERTAD object
E_PAGOMATRICULAPROPIO object
   F_TIENECOMPUTADOR object
   F_TIENEINTERNET.1 object
    F_EDUCACIONMADRE object
  RENDIMIENTO_GLOBAL object
         INDICADOR_1 float64
         INDICADOR_2 float64
         INDICADOR_3 float64
         INDICADOR_4 float64


Podemos ver que únicamente las columnas de indicadores estadísticos del 1-4 representan variables continuas, mientras que el resto representan variables categóricas o intervalos.

Si bien la variable "Período Académico" se representa por un número, sabemos que en realidad este número se asocia más a una fecha que a un valor numérico.

## Valores faltantes

In [20]:
missing_values = training_data.isna().sum()
missing_values[missing_values!=0]

E_VALORMATRICULAUNIVERSIDAD     6287
E_HORASSEMANATRABAJA           30857
F_ESTRATOVIVIENDA              32137
F_TIENEINTERNET                26629
F_EDUCACIONPADRE               23178
F_TIENELAVADORA                39773
F_TIENEAUTOMOVIL               43623
E_PAGOMATRICULAPROPIO           6498
F_TIENECOMPUTADOR              38103
F_TIENEINTERNET.1              26629
F_EDUCACIONMADRE               23664
dtype: int64

In [63]:
print(f"Filas con valores nulos: {missing_values[missing_values!=0].shape[0]}")
print(f"Atributo con mayor número de valores nulos: {missing_values[missing_values!=0].sort_values(ascending=False).index[0]}, con un total de {max(missing_values[missing_values!=0])}")

Filas con valores nulos: 11
Atributo con mayor número de valores nulos: F_TIENEAUTOMOVIL, con un total de 43623


Podemos observar que de las 21 columnas presentes en el modelo, 11 de ellas presentan valores nulos y la columna con mayor cantidad de valores nulos es F_TIENEAUTOMOVIL con un total de 43623 registros cuyo valor es nulo, frente a 692500 registros totales.